kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PropertiesController.java

```diff
@@ -183,17 +282,26 @@ private void setupDescriptionNamePane(EvtType eventType) {
     private void updateDefaultSelectedViews() {
         // default to selected tab (History)
         Toggle tab = propertyToggleButtonGroup.getSelectedToggle();
-        if (editButton.equals(tab)) {
+        if (addEditButton.equals(tab)) {
             contentBorderPane.setCenter(null);
         }
     }
 
     @FXML
-    private void showEditView(ActionEvent event) {
+    private void showAddEditView(ActionEvent event) {
+        LOG.info("Show Add/Edit View " + event);
         event.consume();
-        this.editButton.setSelected(true);
-        LOG.info("Show Edit View " + event);
-        contentBorderPane.setCenter(patternDefinitionPane);
+        this.addEditButton.setSelected(true);
+        contentBorderPane.setCenter(currentEditPane);
+        if (currentEditPane == descriptionNameChooserPane) {
+            // if we will display EDIT for either FQN or Other Name, the display EDIT, otherwise ADD
+            if ((boolean) patternPropertiesViewModel.getPropertyValue(DISPLAY_FQN_EDIT_MODE)
+                    || (boolean) patternPropertiesViewModel.getPropertyValue(DISPLAY_OTHER_NAME_EDIT_MODE)) {
+                this.addEditButton.setText("EDIT");
+            } else {
+                this.addEditButton.setText("ADD");
+            }
```

Comment:
Can we create a Bindings in the `initialize()` method?
```java
BooleanProperty fqnEditMode = patternPropertiesViewModel.getProperty(DISPLAY_FQN_EDIT_MODE);
BooleanProperty otherNameEditMode = patternPropertiesViewModel.getProperty(DISPLAY_OTHER_NAME_EDIT_MODE);

StringProperty buttonText = Bindings.when(fqnEditMode.or(otherNameEditMode).then("Edit").otherwise("Add"));
buttonText.addListener( (obs, oldVal, newVal) -> addEditButton.setText(buttonText.get()));
```

Reply:
there are a couple issues with this...

```
        BooleanProperty isEditModeProp = patternPropertiesViewModel.getProperty(DISPLAY_DEFINITION_EDIT_MODE);
        BooleanProperty fqnEditMode = patternPropertiesViewModel.getProperty(DISPLAY_FQN_EDIT_MODE);
        BooleanProperty otherNameEditMode = patternPropertiesViewModel.getProperty(DISPLAY_OTHER_NAME_EDIT_MODE);
        //FIXME how to handle EDIT vs ADD when what is displayed by the toggle button depends on which bump out...
        //for example, I can fill out the definitions, click that top pencil again and the definitions will have the toggle display EDIT
        //but if I navigate to either of the FQN or OT screens they will be ADD
        //the same problem is compounded when I navigate from the properties toggle at the top and go through the button navigation
        //the FQN will go from ADD to EDIT while the OT will be in ADD before I've filled out at least one
        StringBinding modeText = Bindings.when(isEditModeProp).then("EDIT").otherwise("ADD");
        addEditButton.textProperty().bind(modeText);
```

Reply:
per conversation with @carldea we will potentially refactor this in the future to use a state machine and left it up to me to decide how much property binding logic I want to incorporate.  Since the property binding logic will get more complex here (as in corporates other scenarios/paths through the UI) and we will most likely refactor, I am deciding to leave this alone.